library/Versions.re

```diff
@@ -145,3 +146,20 @@ let getRemoteVersions = () => {
      )
   |> Lwt.return;
 };
+
+let throwIfInstalled = versionName => {
+  switch(getInstalledVersions()) {
+  | Ok(x) => {
+    Array.to_list(x)
+    |> List.exists(x => {
```

Comment:
You can use `Array.exists` too, so no conversion is needed

Reply:
yes...I know `Array.exists` but I rewrote with it like below, compiler said type error. :cry: 

How should I fix it?
```ocaml
    Array.exists(x => {
      open Local;
      x.name == versionName;
    })
    |> x => switch x {
    | true => Lwt.fail(Already_installed(versionName));
    | false => Lwt.return();
    };
```
```
 ERROR  library/Versions.re:158 4-10

155 ┆     x.name == versionName;
156 ┆   })
157 ┆   |> x => switch x {
158 ┆   | true => Lwt.fail(Already_installed(versionName));
159 ┆   | false => Lwt.return();
160 ┆   };
161 ┆ };

This pattern type doesn't match what is expected.

  This type: bool
  Expecting: array(Local.t) => bool

```

Reply:
I'm going to need more context 😄 

What about the following:

```reason
Array.exists(x => {
  open Local;
  x.name == versionName
}, x)

// or

x |> Array.exists(x => {
  open Local;
  x.name == versionName
})
```

Usually, when the compiler starts throwing stuff at me that I don't understand, I'm starting to break down my code to "compilable" pieces: trying to break down things and extract variables. Using `x` as names and piping stuff to anonymous functions tend to be confusing so I'd rather avoid it unless it is a one-liner or something really general

Reply:
Thank you! I understood it! :heart: